docker-entrypoint.sh

```diff
@@ -19,6 +19,32 @@ run_as() {
     fi
 }
 
+# usage: file_env VAR [DEFAULT]
+#    ie: file_env 'XYZ_PASSWORD' 'example'
+# (will allow for "$XYZ_PASSWORD_FILE" to fill in the value of
+#  "$XYZ_PASSWORD" from a file, especially for Docker's secrets feature)
+file_env() {
+	local var="$1"
+	local fileVar="${var}_FILE"
+	local def="${2:-}"
+	if [ "${!var:-}" ] && [ "${!fileVar:-}" ]; then
+		echo >&2 "error: both $var and $fileVar are set (but are exclusive)"
+		exit 1
+	fi
+	local val=""
+	if [ "${!var:-}" ]; then
+		val="${!var}"
+	elif [ "${!fileVar:-}" ]; then
+		val="$(< "${!fileVar}")"
+    	unset "$fileVar"
```

Comment:
code style

Reply:
Where can I read about recommended code style?

Reply:
Just use the same code style in the whole file.
The `unset` statement is either not intended correctly or it should be placed after the conditions, near `export "$var"="$val"`.

Reply:
`unset` used to remove `*_PASSWORD_FILE` variable when it defined. Why should I unset this variable while it is not defined?

Reply:
This block of code got from [official mariadb Dockerfile](https://github.com/docker-library/mariadb/blob/master/10.3/docker-entrypoint.sh#L21-L41)

Reply:
The line indention there looks correctly.